homeassistant/components/pi_hole/sensor.py

```diff
@@ -64,7 +71,32 @@ def __init__(
     @property
     def native_value(self) -> Any:
         """Return the state of the device."""
+        if self.entity_description.key == "available_updates":
+            available_updates = 0
+            if self.api.versions["core_update"]:
+                available_updates += 1
+            if self.api.versions["web_update"]:
+                available_updates += 1
+            if self.api.versions["FTL_update"]:
+                available_updates += 1
+            return available_updates
```

Comment:
I recommend instead creating a binary sensor for each update type with device class `update`.

https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes

Put the current and latest versions as state attributes on each binary sensor.